SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -183,11 +183,11 @@ void USpatialPackageMapClient::ResolveSubobject(UObject* Object, const FUnrealOb
 	}
 	else
 	{
-		// TODO UNR-5785 - Remove this once fixed, as really SpatialGuidCache and Native's GuidCache should be in sync
+		// Sanity check consistency between SpatialGuidCache and native's GuidCache, as it would be pretty bad if they were out of sync.
 		const FNetworkGUID ObjectNetGUID = GetNetGUIDFromObject(Object);
 		if (!ObjectNetGUID.IsValid())
 		{
-			UE_LOG(LogSpatialPackageMap, Log,
+			UE_LOG(LogSpatialPackageMap, Warning,
 				   TEXT("SpatialGuidCache has a NetGUID mapping which native's GuidCache does not have for object: %s with NetGUID: %u. "
 						"Removing existing mapping from SpatialGuidCache and recreating."),
 				   *Object->GetName(), NetGUID.Value);
```

Comment:
Do we think this sanity check and making consistent again stuff is worth keeping? Maybe could remove the making consistent again lines (removesubobject and then SpatialGuidCache->AssignNewSubobjectNetGUID) but change the warning to error?

Reply:
As I am not sure if this could happen with some weirdness coming from async loading, I would keep it at warning and keep the repair stuff in.
Hmmm, I would like to create some kind of reminders for us to remove this code if we see no reports of the warning in like a year or so.

Reply:
like a ticket that will ping us in a year? Or just a note in the code saying `Remove this error handling if after 07/2022 and we haven't experienced the warning`

Reply:
This would be a perfect match for always on event tracing, but we don't do that sadly.